app/src/main/java/com/davidtomas/taskyapp/features/auth/presentation/register/components/Header.kt

```diff
@@ -0,0 +1,86 @@
+package com.davidtomas.taskyapp.features.auth.presentation.register.components
+
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import com.davidtomas.taskyapp.R
+import com.davidtomas.taskyapp.core.domain.util.EMPTY_STRING
+import com.davidtomas.taskyapp.coreUi.LocalSpacing
+
+@Composable
+fun Header(
+    modifier: Modifier = Modifier,
+    headerText: String = String.EMPTY_STRING,
+    @DrawableRes leadingIcon: Int? = null,
+    onLeadingIconClick: (() -> Unit)? = null,
```

Comment:
Could it make sense to be able to pass a custom composable lambda here for the icon? Right now, you wouldn't even be able to pass material icons

Reply:
I thought about it and is a good point. How would you do to have the same Composable with three different signatures? One for passing only a drawable, another one for passing a custom composable and another one to pass a material Icon? I think is the best way to have the most customizable coposable and can be reused in different apps. In order to do that it would requier triplicating the code and making the necessary minor changes?

Reply:
I'd see the point of 2 overloads, but not 3. One could take the lambda and one an `ImageVector`. For image vectors, you can pass both material icons and your own ones (`ImageVector.vectorResource(R.drawable.x)`).

And you don't need to duplicate the code for that, you just need to provide another overload and call the other function in it